bson/src/main/org/bson/codecs/pojo/PojoBuilderHelper.java

```diff
@@ -157,7 +157,7 @@
                                                                   final List<String> genericTypeNames,
                                                                   final Type genericType) {
         PropertyMetadata<T> propertyMetadata = getOrCreatePropertyMetadata(propertyName, declaringClassName, propertyNameMap, typeData);
-        if (!propertyMetadata.getTypeData().getType().isAssignableFrom(typeData.getType())) {
+        if (!isAssignableClass(propertyMetadata.getTypeData().getType(), typeData.getType())) {
```

Comment:
Because we don't know if the originating model contains a property that has a stricter or looser class type, we have to accept either.